crates/core/src/pattern/patterns.rs

```diff
@@ -562,6 +562,26 @@ impl Pattern {
                     ))
                 })
                 .collect::<Result<Vec<(u16, bool, Pattern)>>>()?;
+            let mut mandatory_empty_args = fields.iter().filter(|field| {
+                node.child_by_field_id(field.id()).is_none() && lang.mandatory_empty_field(sort, field.id())
+            }).map(|field| {
+                if field.multiple() {
+                    (
+                        field.id(),
+                        true,
+                        Pattern::List(Box::new(List::new(Vec::new()))))
+                } else {
+                    (
+                        field.id(),
+                        false,
+                        Pattern::Dynamic(DynamicPattern::Snippet(DynamicSnippet {
+                        parts: vec![DynamicSnippetPart::String("".to_string())],
```

Comment:
Isn't this special casing it as a `""`? Not all fields would have an empty string as a valid option, let alone using this syntax.

Why not use `Constant::null` instead?

Reply:
This mirrors what we use for the `.` pattern. I think having a proper dot pattern would probably be better, but I don't think it's worth changing here.

Reply:
Ok, we did have https://github.com/getgrit/gritql/blob/0cb34abd869ee9121d7689c77f3a0ede2ea4c21e/crates/core/src/pattern/patterns.rs#L98 but this is fine?